jenkins/github_helpers.py

```diff
@@ -71,6 +75,29 @@ def connect_to_repo(self, github_instance, repo_name):
             "credentials and try again.".format(repo_name)
         )
 
+    def repo_from_remote(self, repo_root, remote_name_filter=None):
+        """
+        Get the repository object for a repository with a Github remote.
+
+        Optionally restrict the remotes under consideration by passing a list
+        of names as``remote_name_filter``.
```

Comment:
1. Consider providing an example list of names.
2. Maybe `remote_name_allow_list` would make it more clear that you are filtering in, and not filtering out.